### PR TITLE
fix date_as_argument feature to add timestamp argument at every cron …

### PR DESCRIPTION
**Pull request by vsuhachev**
At Readme `date_as_argument` described as
```
  'date_as_argument' => true, # add the time of execution as last argument of the perform method
```
It is implemented as "add the time of ~~execution~~creation of schedule as last argument of the perform method". It is confusing and mostly useless. If I want to add the creation time of the schedule, then it's easier for me to do it explicitly from the client code.

This PR implements `date_as_argument` feature to add timestamp as extra argument on every job execution

### CODE REVIEW

**Comment by markets** [comment]
Hi @vsuhachev and thanks for sending this PR! 

It seems ok to me, but since I'm still not familiar with the code base internals, I'd like to wait for feedback from other contributors.

CI is broken due to the Sidekiq upgrade, if you update your branch from current `master`, this will be fixed.

**Comment by markets** [comment]
@ondrejbartas @honzasterba are you agree with this one? Not sure which was the initial idea of that `date_as_argument` attribute

File: .gitignore
Comment (honzasterba) [inline .gitignore:15]: please remove this, this is system specific and should be in your global .gitignore and the change is not related to the PR

**Review by honzasterba — COMMENTED** [review]
the code looks good to me

the PR description needs to be updated to better document in what situation this behaviour is not working and why this fix is necessary

preferably that would be obvious from the tests, but it is not to me at least

File: .gitignore
Comment (vsuhachev) [reply .gitignore]: removed

**Comment by markets** [comment]
@honzasterba Could you please make a second review 🙏🏼?

**Review by honzasterba — APPROVED** [review]
LGTM

from the PR description I would think it would be best to remove this feature :)

**Comment by markets** [comment]
Yep, not sure what is the initial idea behind this attribute. We're not using this flag in our projects with `sidekiq-cron` currently, but as per the docs, now it seems "broken" (it does a different thing). 

I'd be ok to completely remove it too 😃, but it seems some people would like it working as described (several +1 in the original issue #247).

**Comment by honzasterba** [comment]
Oh, so people are using it, great lets merge it then!

**Pull request merged**
